src/tools/crossgen2/Common/Compiler/VectorFieldLayoutAlgorithm.cs

```diff
@@ -106,5 +106,10 @@ public static bool IsVectorType(DefType type)
                 type.Name == "Vector128`1" ||
                 type.Name == "Vector256`1");
         }
+
+        public static bool IsVectorOfTType(DefType type)
```

Comment:
For https://github.com/dotnet/coreclr/pull/23899, I had added this as `InteropTypes.IsSystemNumericsVectorT`, as that appears to be where a fair number of the other `IsType` checks live

Reply:
I am not sure which location is better. Your methods are also a bit different as they require the TypeSystemContext parameter. So maybe keeping both versions makes sense. There was already the IsVectorType here, so adding the IsVectorOfTypeT to the same place felt ok.

Reply:
`InteropTypes` only handles types interesting for interop (as the name implies). Up until Tanner's change it was exclusively used in generating/figuring out marshalling. Crossgen2 doesn't have a central `MscorlibBinder` equivalent (as it exists in CoreCLR).

We place these as close to the usage as possible. Centralized locations tend to end up with orphaned entries.